topics/wallet-iteration-standard/ledger-extension.md

```diff
@@ -0,0 +1,91 @@
+# ICRC-27: Ledger extension for Wallet Interaction Standard
+
+| Status |
+|:------:|
+| Draft  |
+
+## Summary
+
+This specification describes a ledger extension for the ICRC-25 Wallet Interaction Standard.
+
+## Terminology
+
+* wallet: A service that manages a user's keys and can sign and perform canister calls on their behalf.
+* relying party: A service that wants to request calls on a specific canister.
+
+## Messages
+
+### `permission`
+
+Additional `ledger` scope is passed in the request and in the response an additional `subaccounts` property is returned in each `identity` when the wallet supports this extension.
+
+There are many types of wallets:
+- Wallets that support many different standards
+- Wallets that support multiple networks
+- Wallets that keep a list of subaccounts per ledger
+- Wallets that keep a list of subaccounts used across all ledgers
+
+To support all these different wallets and pass these details to the relying party, the subaccounts in this spec are defined with multiple properties to indicate on which networks, standards and ledgers the subaccounts are meant to be used.
+
+#### Request
+
+The `ledger` scope is passed to the ICRC-25 `permission` request message scopes, no additional fields in the request are required in this spec.
+
+#### Response
+
+`identities`: Same as ICRC-25 response with additional `ledger` field.
+  - `subaccounts` (`vec`): List of ledger subaccounts for this identity.
```

Comment:
It would be great if we could agree on how to handle the default (all `0`'s) subaccount.
Is it implied if not listed in subaccounts?

Reply:
My first thought is, now that we have this special `ledger` scope to impose more restrictions and awareness, it should be be explicitly listed as in "the user has explicitly agreed to share the default subaccount" 🤔 

Of course the relying party could still access the default account using the principal (derived from the `publicKey`) from the response without any permission, but if it tried to request an action from the wallet with that account, it would have to have the permission.

Reply:
I've updated the standard with a section regarding the default subaccount. Besides the reasons you've mentioned above, a more crucial reason for requiring the default subaccount to be explicitly defined was that some wallets might not support the default subaccount but do support other subaccounts.